Add Nebius Cloud (#59)
outerbounds/metaflow-tools: nebius/terraform/README.md

# A minimal viable Metaflow-on-Nebius stack

## What does it do?

> It provisions all necessary Nebius resources. Main resources are:

Reviewer: ```suggestion
It provisions all necessary Nebius resources. The main resources are:
```

* Nebius S3
* Nebius Mk8s

It will also deploy Metaflow services onto the Mk8s cluster above.

## Prerequisites

* Install [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli).
* Install [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl).

> * Install [nebius](https://docs.nebius.com/cli/quickstart/) CLI and setup it.

Reviewer: ```suggestion
* Install [nebius](https://docs.nebius.com/cli/quickstart/) CLI and set it up.
```

## Usage

The templates are organized into two modules, `infra` and `services`.

Before you do anything, create a TF vars file `FILE.tfvars` (`FILE` could be something else), with this content.

```text
tenant_id = "" # you can get from your console
project_id = "" # you can get from your console
```

This is used to help generate unique resource names for:

* MPG server name
* S3 storage account name

Note: these resources must be globally unique across all of Nebius.

Next run to obtain Iam Nebius token

```bash
. ./environment.sh
```

Next, apply the `infra` module (creates Nebius cloud resources only).

> ```bash

Reviewer: ```suggestion
terraform init
```

Need to mention terraform init somewhere as well

terraform apply -target="module.infra" -var-file=FILE.tfvars
```

> Next, add Service Account to `editors` group and add values to `FILE.tfvars`. You can get values from console from SA page

Reviewer: ```suggestion
Next, add the Service Account to the `editors` group and add values to `FILE.tfvars`. You can get values from the console from the Service Account page:
```

# A minimal viable Metaflow-on-Nebius stack

## What does it do?

It provisions all necessary Nebius resources. The main resources are:

* Nebius S3
* Nebius Mk8s

It will also deploy Metaflow services onto the Mk8s cluster above.

## Prerequisites

* Install [terraform](https://learn.hashicorp.com/tutorials/terraform/install-cli).
* Install [kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl).
* Install [nebius](https://docs.nebius.com/cli/quickstart/) CLI and set it up.

## Usage

The templates are organized into two modules, `infra` and `services`.

Before you do anything, create a TF vars file `FILE.tfvars` (`FILE` could be something else), with this content.

```text
tenant_id = "" # you can get from your console

> project_id = "" # you can get from your console

Reviewer: Can you add instructions for `vpc_subnet_id` too

```

This is used to help generate unique resource names for:

* MPG server name
* S3 storage account name

Note: these resources must be globally unique across all of Nebius.

Next run to obtain Iam Nebius token

```bash
. ./environment.sh
```

Next, apply the `infra` module (creates Nebius cloud resources only).

```bash
terraform init  
terraform apply -target="module.infra" -var-file=FILE.tfvars
```

Next, add the Service Account to the `editors` group and add values to `FILE.tfvars`. You can get values from the console from the Service Account page:

```txt
aws_access_key_id = ""
aws_secret_access_key=""
```

Then run:
```bash
terraform apply -target="module.services" -var-file=FILE.tfvars
```

> 

Reviewer: Can you also add a section on how to destroy the infra?
